numpy/core/include/numpy/ndarraytypes.h

```diff
@@ -1894,6 +1898,8 @@ typedef void (PyDataMem_EventHookFunc)(void *inp, void *outp, size_t size,
         discover_descr_from_pyobject_function *discover_descr_from_pyobject;
         is_known_scalar_type_function *is_known_scalar_type;
         default_descr_function *default_descr;
+        common_dtype_function *common_dtype;
+        common_instance_function *common_instance;
     };
 
 #endif  /* NPY_INTERNAL_BUILD */
```

Comment:
In a follow-on PR it would be nice to move this out of public headers, it is guarded by `NPY_INTERNAL_BUILD` anyway.